test/compress/destructuring.js

```diff
@@ -200,3 +200,13 @@ destructuring_remove_unused_2: {
         }
     }
 }
+
+object_destructuring_may_need_parentheses: {
+    beautify = {
+        ecma: 6
+    }
+    input: {
+        ({a, b} = {a: 1, b: 2});
```

Comment:
`acorn@4.0.4`, esprima, and `espree` accept it while `babylon` and `traceur` do not:

https://astexplorer.net/#/Qsed8hCrRx